### ISSUE TITLE
File unparseable in `redshift`, passes in `postgres`

### ISSUE THREAD

**Issue by puetsch** [sqlfluff/sqlfluff#1667]
```sql
select
      id
from table1
where (idle_time_minutes > 30 or idle_time_minutes is null)
```

## Expected Behaviour
passes

## Observed Behaviour
```
=== [ path: test.sql ] ===

WARNING    Unable to lint test.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/linter/runner.py", line 97, in run
    yield partial()
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/linter/linter.py", line 521, in lint_rendered
    parsed = cls.parse_rendered(rendered)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/linter/linter.py", line 312, in parse_rendered
    tokens, rendered.config, recurse=recurse, fname=rendered.fname
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/linter/linter.py", line 189, in _parse_tokens
    tokens, recurse=recurse, fname=fname
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/parser.py", line 32, in parse
    parsed = root_segment.parse(parse_context=ctx)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/base.py", line 870, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/base.py", line 276, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/base.py", line 870, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/base.py", line 276, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/base.py", line 870, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/base.py", line 276, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/base.py", line 811, in parse
    m = parse_grammar.match(segments=segments, parse_context=ctx)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 130, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 198, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 146, in _match_once
    trim_noncode=False,
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/base.py", line 218, in _longest_trimmed_match
    segments, parse_context=parse_context
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 331, in match
    content_match = super().match(content_segs, parse_context=ctx)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 130, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/base.py", line 854, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/base.py", line 466, in match
    m = cls.match_grammar.match(segments=segments, parse_context=ctx)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/base.py", line 854, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 130, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 198, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 146, in _match_once
    trim_noncode=False,
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/base.py", line 218, in _longest_trimmed_match
    segments, parse_context=parse_context
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 198, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 146, in _match_once
    trim_noncode=False,
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/base.py", line 218, in _longest_trimmed_match
    segments, parse_context=parse_context
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 130, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/base.py", line 854, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 198, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 134, in _match_once
    segments, parse_context=parse_context
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 72, in _prune_options
    simple = opt.simple(parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/base.py", line 63, in wrapped_method
    result = func(self, parse_context=parse_context)
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/base.py", line 783, in simple
    return self._get_elem(dialect=parse_context.dialect).simple(
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/parser/grammar/base.py", line 805, in _get_elem
    return dialect.ref(self._get_ref())
  File "/opt/anaconda3/lib/python3.7/site-packages/sqlfluff/core/dialects/base.py", line 234, in ref
    name, self.name
RuntimeError: Grammar refers to 'NanKeywordSegment' which was not found in the redshift dialect
```

## Steps to Reproduce
run `sqlfluff lint --dialect=redshift -v` on file above

Note: it passes in `postgres`

## Dialect
redshift

## Version
0.7.0 on Python 3.7.6

## Configuration
```
[sqlfluff]
verbose = 0
nocolor = False
dialect = postgres
templater = jinja
rules = None
exclude_rules = L009,L011,L019,L031,L034
recurse = 0
output_line_length = 160
runaway_limit = 10
ignore_templated_areas = True
encoding = autodetect
# Comma separated list of file extensions to lint.

# NB: This config will only apply in the root folder.
sql_file_exts = .sql,.sql.j2,.dml,.ddl

[sqlfluff:indentation]
indented_joins = False
indented_using_on = True
template_blocks_indent = True

[sqlfluff:templater]
unwrap_wrapped_queries = True

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

[sqlfluff:templater:jinja:macros]
# Macros provided as builtins for dbt projects
dbt_ref = {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
dbt_source = {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
dbt_config = {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
dbt_var = {% macro var(variable) %}item{% endmacro %}
dbt_is_incremental = {% macro is_incremental() %}True{% endmacro %}

# Some rules can be configured directly from the config common to other rules.
[sqlfluff:rules]
tab_space_size = 4
max_line_length = 120
indent_unit = space
comma_style = trailing
allow_scalar = True
single_table_references = consistent
unquoted_identifiers_policy = all

# Some rules have their own specific config.
[sqlfluff:rules:L003]
lint_templated_tokens = True

[sqlfluff:rules:L007]  # Keywords
operator_new_lines = after

[sqlfluff:rules:L010]  # Keywords
capitalisation_policy = consistent

[sqlfluff:rules:L014]  # Unquoted identifiers
extended_capitalisation_policy = consistent

[sqlfluff:rules:L016]
ignore_comment_lines = False

[sqlfluff:rules:L029]  # Keyword identifiers
unquoted_identifiers_policy = aliases

[sqlfluff:rules:L030]  # Function names
capitalisation_policy = consistent

[sqlfluff:rules:L038]
select_clause_trailing_comma = forbid

[sqlfluff:rules:L040]  # Null & Boolean Literals
capitalisation_policy = consistent

[sqlfluff:rules:L042]
# By default, allow subqueries in from clauses, but not join clauses.
forbid_subquery_in = join

[sqlfluff:rules:L047]  # Consistent syntax to count all rows
prefer_count_1 = False
prefer_count_0 = False

```
